ICE caused by using feature(generic_const_exprs)

<!--
Thank you for finding an Internal Compiler Error! 🧊  If possible, try to provide
a minimal verifiable example. You can read "Rust Bug Minimization Patterns" for
how to create smaller examples.
http://blog.pnkfx.org/blog/2019/11/18/rust-bug-minimization-patterns/
-->

### Code
The code runs as expected until we introduce the `generic_const_exprs` attribute and the compiler encounters an ICE error.
```Rust
#![feature(generic_const_exprs)]
fn bug<'a>() {
    [(); (|_: &'a u8| (), 0).1];
}
```


### Meta
<!--
If you're using the stable version of the compiler, you should also check if the
bug also exists in the beta or nightly versions.
-->

`rustc --version --verbose`:
```
$ rustc --version --verbose
rustc 1.73.0-nightly (474709a9a 2023-08-03)
binary: rustc
commit-hash: 474709a9a2a74a8bcf0055fadb335d0ca0d2d939
commit-date: 2023-08-03
host: x86_64-unknown-linux-gnu
release: 1.73.0-nightly
LLVM version: 16.0.5
```

### Error output

```
warning: the feature `generic_const_exprs` is incomplete and may not be safe to use and/or cause compiler crashes
 --> file1.rs:1:12
  |
1 | #![feature(generic_const_exprs)]
  |            ^^^^^^^^^^^^^^^^^^^
  |
  = note: see issue #76560 <https://github.com/rust-lang/rust/issues/76560> for more information
  = note: `#[warn(incomplete_features)]` on by default

error[E0601]: `main` function not found in crate `file1`
 --> file1.rs:4:2
  |
4 | }
  |  ^ consider adding a `main` function to `file1.rs`

error: internal compiler error: compiler/rustc_borrowck/src/universal_regions.rs:880:36: cannot convert `ReFree(DefId(0:3 ~ file1[e613]::bug), BrNamed(DefId(0:4 ~ file1[e613]::bug::'a), 'a))` to a region vid

thread 'rustc' panicked at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/compiler/rustc_errors/src/lib.rs:1645:9:
Box<dyn Any>
stack backtrace:
   0:     0x7f7803f3c19c - std::backtrace_rs::backtrace::libunwind::trace::haf256adafafbe58d
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f7803f3c19c - std::backtrace_rs::backtrace::trace_unsynchronized::hfe1951132ff691c0
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f7803f3c19c - std::sys_common::backtrace::_print_fmt::h9a0fe52434930c36
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x7f7803f3c19c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h24adccbf3e1ada83
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f7803fa289c - core::fmt::rt::Argument::fmt::h18ba555e398addfe
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/core/src/fmt/rt.rs:138:9
   5:     0x7f7803fa289c - core::fmt::write::hbf3ee2d80be74759
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/core/src/fmt/mod.rs:1094:21
   6:     0x7f7803f2e90e - std::io::Write::write_fmt::hfc2b3251522ff943
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/io/mod.rs:1714:15
   7:     0x7f7803f3bf85 - std::sys_common::backtrace::_print::hfdc8ddb5f3ddee36
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x7f7803f3bf85 - std::sys_common::backtrace::print::h316f264b298c7a30
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x7f7803f3f0da - std::panicking::panic_hook_with_disk_dump::{{closure}}::h136bbb963feeea4a
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/panicking.rs:278:22
  10:     0x7f7803f3ed73 - std::panicking::panic_hook_with_disk_dump::hd91a018a982a84f7
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/panicking.rs:312:9
  11:     0x7f7802d0d799 - rustc_driver_impl[2c2bb0da6cfb3826]::install_ice_hook::{closure#0}
  12:     0x7f7803f3f980 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::hd99b1fe24ccd07b5
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/alloc/src/boxed.rs:2021:9
  13:     0x7f7803f3f980 - std::panicking::rust_panic_with_hook::h1593161995c9c003
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/panicking.rs:733:13
  14:     0x7f78032971f1 - std[b7e5a9c79ea9fbed]::panicking::begin_panic::<rustc_errors[4345527545ed2164]::ExplicitBug>::{closure#0}
  15:     0x7f7803296096 - std[b7e5a9c79ea9fbed]::sys_common::backtrace::__rust_end_short_backtrace::<std[b7e5a9c79ea9fbed]::panicking::begin_panic<rustc_errors[4345527545ed2164]::ExplicitBug>::{closure#0}, !>
  16:     0x7f780322afc6 - std[b7e5a9c79ea9fbed]::panicking::begin_panic::<rustc_errors[4345527545ed2164]::ExplicitBug>
  17:     0x7f780320b344 - <rustc_errors[4345527545ed2164]::HandlerInner>::bug::<alloc[b2a4fa84f81d55c8]::string::String>
  18:     0x7f780320b1d6 - <rustc_errors[4345527545ed2164]::Handler>::bug::<alloc[b2a4fa84f81d55c8]::string::String>
  19:     0x7f780328b20c - rustc_middle[32728bc096d751c7]::util::bug::opt_span_bug_fmt::<rustc_span[9abd768efbccd329]::span_encoding::Span>::{closure#0}
  20:     0x7f780328a00a - rustc_middle[32728bc096d751c7]::ty::context::tls::with_opt::<rustc_middle[32728bc096d751c7]::util::bug::opt_span_bug_fmt<rustc_span[9abd768efbccd329]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  21:     0x7f7803289fd8 - rustc_middle[32728bc096d751c7]::ty::context::tls::with_context_opt::<rustc_middle[32728bc096d751c7]::ty::context::tls::with_opt<rustc_middle[32728bc096d751c7]::util::bug::opt_span_bug_fmt<rustc_span[9abd768efbccd329]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  22:     0x7f780165a400 - rustc_middle[32728bc096d751c7]::util::bug::bug_fmt
  23:     0x7f7801e8d860 - <rustc_borrowck[36619a147d0794a8]::type_check::constraint_conversion::ConstraintConversion>::convert_all
  24:     0x7f7801e8e1c6 - <rustc_borrowck[36619a147d0794a8]::type_check::TypeChecker>::push_region_constraints
  25:     0x7f7801a5ab80 - <rustc_borrowck[36619a147d0794a8]::type_check::TypeChecker>::ascribe_user_type_skip_wf
  26:     0x7f7801a53c90 - rustc_borrowck[36619a147d0794a8]::type_check::type_check
  27:     0x7f7801a4a34f - rustc_borrowck[36619a147d0794a8]::nll::compute_regions
  28:     0x7f7801a24ed5 - rustc_borrowck[36619a147d0794a8]::do_mir_borrowck
  29:     0x7f7801a231e5 - rustc_borrowck[36619a147d0794a8]::mir_borrowck
  30:     0x7f7800d3976e - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>
  31:     0x7f7800d3973e - <rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_span[9abd768efbccd329]::def_id::LocalDefId)>>::call_once
  32:     0x7f7800e80c3d - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::VecCache<rustc_span[9abd768efbccd329]::def_id::LocalDefId, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  33:     0x7f780269094f - rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
  34:     0x7f7801e9cb9f - <rustc_borrowck[36619a147d0794a8]::type_check::TypeChecker>::prove_closure_bounds
  35:     0x7f7801e6b39d - <rustc_borrowck[36619a147d0794a8]::type_check::TypeChecker>::typeck_mir
  36:     0x7f7801a52a01 - rustc_borrowck[36619a147d0794a8]::type_check::type_check
  37:     0x7f7801a4a34f - rustc_borrowck[36619a147d0794a8]::nll::compute_regions
  38:     0x7f7801a24ed5 - rustc_borrowck[36619a147d0794a8]::do_mir_borrowck
  39:     0x7f7801a231e5 - rustc_borrowck[36619a147d0794a8]::mir_borrowck
  40:     0x7f7800d3976e - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>
  41:     0x7f7800d3973e - <rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_span[9abd768efbccd329]::def_id::LocalDefId)>>::call_once
  42:     0x7f7800e80c3d - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::VecCache<rustc_span[9abd768efbccd329]::def_id::LocalDefId, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  43:     0x7f780269094f - rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
  44:     0x7f780137344e - rustc_mir_transform[2797bcf166cffb39]::mir_drops_elaborated_and_const_checked
  45:     0x7f7800d42a3e - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::mir_drops_elaborated_and_const_checked::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>
  46:     0x7f7800d42a0e - <rustc_query_impl[c311561e1f786c65]::query_impl::mir_drops_elaborated_and_const_checked::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_span[9abd768efbccd329]::def_id::LocalDefId)>>::call_once
  47:     0x7f7800e80c3d - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::VecCache<rustc_span[9abd768efbccd329]::def_id::LocalDefId, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  48:     0x7f780268752f - rustc_query_impl[c311561e1f786c65]::query_impl::mir_drops_elaborated_and_const_checked::get_query_non_incr::__rust_end_short_backtrace
  49:     0x7f7801bf4b63 - rustc_mir_transform[2797bcf166cffb39]::mir_for_ctfe
  50:     0x7f7800dabe4e - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::mir_for_ctfe::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>
  51:     0x7f7800dabe21 - <rustc_query_impl[c311561e1f786c65]::query_impl::mir_for_ctfe::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_span[9abd768efbccd329]::def_id::DefId)>>::call_once
  52:     0x7f7800d38145 - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::DefaultCache<rustc_span[9abd768efbccd329]::def_id::DefId, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  53:     0x7f7802687680 - rustc_query_impl[c311561e1f786c65]::query_impl::mir_for_ctfe::get_query_non_incr::__rust_end_short_backtrace
  54:     0x7f7801279c60 - <rustc_const_eval[9e95d3ca9e32f8a5]::interpret::eval_context::InterpCx<rustc_const_eval[9e95d3ca9e32f8a5]::const_eval::machine::CompileTimeInterpreter>>::load_mir
  55:     0x7f780117f901 - rustc_const_eval[9e95d3ca9e32f8a5]::const_eval::eval_queries::eval_to_allocation_raw_provider
  56:     0x7f7800d8a427 - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 16usize]>>
  57:     0x7f7801c219f7 - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::DefaultCache<rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  58:     0x7f7801c21679 - rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_allocation_raw::get_query_non_incr::__rust_end_short_backtrace
  59:     0x7f780117fe67 - rustc_const_eval[9e95d3ca9e32f8a5]::const_eval::eval_queries::eval_to_allocation_raw_provider
  60:     0x7f7800d8a427 - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 16usize]>>
  61:     0x7f7801c219f7 - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::DefaultCache<rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  62:     0x7f7801c21679 - rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_allocation_raw::get_query_non_incr::__rust_end_short_backtrace
  63:     0x7f780246503a - <rustc_const_eval[9e95d3ca9e32f8a5]::provide::{closure#0} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>)>>::call_once
  64:     0x7f780217a459 - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_valtree::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 24usize]>>
  65:     0x7f780217a408 - <rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_valtree::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>)>>::call_once
  66:     0x7f78022fa0c8 - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::DefaultCache<rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 24usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  67:     0x7f78022f9d5e - rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_valtree::get_query_non_incr::__rust_end_short_backtrace
  68:     0x7f78012eab4c - rustc_middle[32728bc096d751c7]::query::plumbing::query_get_at::<rustc_query_system[1215f1fc07ea991a]::query::caches::DefaultCache<rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 24usize]>>>
  69:     0x7f78012ea8eb - <rustc_middle[32728bc096d751c7]::ty::context::TyCtxt>::const_eval_global_id_for_typeck
  70:     0x7f78012ea3c4 - <rustc_middle[32728bc096d751c7]::ty::context::TyCtxt>::const_eval_resolve_for_typeck
  71:     0x7f780145ca87 - <rustc_middle[32728bc096d751c7]::ty::consts::Const>::eval
  72:     0x7f78013e71ff - <rustc_hir_typeck[a1aa762a977bcd63]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  73:     0x7f7801425d54 - <rustc_hir_typeck[a1aa762a977bcd63]::fn_ctxt::FnCtxt>::check_block_with_expected
  74:     0x7f78013e3433 - <rustc_hir_typeck[a1aa762a977bcd63]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  75:     0x7f780156899b - <rustc_hir_typeck[a1aa762a977bcd63]::fn_ctxt::FnCtxt>::check_return_expr
  76:     0x7f780156704c - rustc_hir_typeck[a1aa762a977bcd63]::check::check_fn
  77:     0x7f7801555b55 - rustc_hir_typeck[a1aa762a977bcd63]::typeck
  78:     0x7f7800d0b08c - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::typeck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>
  79:     0x7f7800d0b06e - <rustc_query_impl[c311561e1f786c65]::query_impl::typeck::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_span[9abd768efbccd329]::def_id::LocalDefId)>>::call_once
  80:     0x7f7800e80c3d - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::VecCache<rustc_span[9abd768efbccd329]::def_id::LocalDefId, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  81:     0x7f78026906cf - rustc_query_impl[c311561e1f786c65]::query_impl::typeck::get_query_non_incr::__rust_end_short_backtrace
  82:     0x7f780219be22 - rustc_data_structures[e5ddef51fa833839]::sync::par_for_each_in::<&[rustc_span[9abd768efbccd329]::def_id::LocalDefId], <rustc_middle[32728bc096d751c7]::hir::map::Map>::par_body_owners<rustc_hir_analysis[505bd014cb0020b8]::check_crate::{closure#7}>::{closure#0}>
  83:     0x7f780219ae2a - rustc_hir_analysis[505bd014cb0020b8]::check_crate
  84:     0x7f78021955ca - rustc_interface[b10bd5cb1b971de9]::passes::analysis
  85:     0x7f78021ffd8a - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 1usize]>>
  86:     0x7f78021ffd79 - <rustc_query_impl[c311561e1f786c65]::query_impl::analysis::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, ())>>::call_once
  87:     0x7f780239b368 - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::SingleCache<rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  88:     0x7f780239b0f7 - rustc_query_impl[c311561e1f786c65]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  89:     0x7f7801f798c5 - <rustc_middle[32728bc096d751c7]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[2c2bb0da6cfb3826]::run_compiler::{closure#1}::{closure#2}::{closure#4}, core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>>
  90:     0x7f7801f78ef2 - <rustc_interface[b10bd5cb1b971de9]::interface::Compiler>::enter::<rustc_driver_impl[2c2bb0da6cfb3826]::run_compiler::{closure#1}::{closure#2}, core[8e2a8f22f3da315]::result::Result<core[8e2a8f22f3da315]::option::Option<rustc_interface[b10bd5cb1b971de9]::queries::Linker>, rustc_span[9abd768efbccd329]::ErrorGuaranteed>>
  91:     0x7f7801f71fc8 - std[b7e5a9c79ea9fbed]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[b10bd5cb1b971de9]::util::run_in_thread_pool_with_globals<rustc_interface[b10bd5cb1b971de9]::interface::run_compiler<core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>, rustc_driver_impl[2c2bb0da6cfb3826]::run_compiler::{closure#1}>::{closure#0}, core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>>
  92:     0x7f7801f7174e - <<std[b7e5a9c79ea9fbed]::thread::Builder>::spawn_unchecked_<rustc_interface[b10bd5cb1b971de9]::util::run_in_thread_pool_with_globals<rustc_interface[b10bd5cb1b971de9]::interface::run_compiler<core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>, rustc_driver_impl[2c2bb0da6cfb3826]::run_compiler::{closure#1}>::{closure#0}, core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>>::{closure#1} as core[8e2a8f22f3da315]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  93:     0x7f7803f4a125 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hb18992c0c074fb04
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/alloc/src/boxed.rs:2007:9
  94:     0x7f7803f4a125 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::ha4998f5f9033c44e
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/alloc/src/boxed.rs:2007:9
  95:     0x7f7803f4a125 - std::sys::unix::thread::Thread::new::thread_start::h3370231efa79af31
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/sys/unix/thread.rs:108:17
  96:     0x7f77ff4c76db - start_thread
  97:     0x7f77ff1f061f - clone
  98:                0x0 - <unknown>

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: please attach the file at `/home/coding/rustc-ice-2023-08-04T08:42:06.45358537Z-23008.txt` to your bug report

query stack during panic:
#0 [mir_borrowck] borrow-checking `bug::{constant#0}::{closure#0}`
#1 [mir_borrowck] borrow-checking `bug::{constant#0}`
#2 [mir_drops_elaborated_and_const_checked] elaborating drops for `bug::{constant#0}`
#3 [mir_for_ctfe] caching mir of `bug::{constant#0}` for CTFE
#4 [eval_to_allocation_raw] const-evaluating + checking `bug::{constant#0}`
#5 [eval_to_allocation_raw] const-evaluating + checking `bug::{constant#0}`
#6 [eval_to_valtree] evaluating type-level constant
#7 [typeck] type-checking `bug`
#8 [analysis] running analysis passes on this crate
end of query stack
error: aborting due to 2 previous errors; 1 warning emitted

For more information about this error, try `rustc --explain E0601`.


```

<!--
Include a backtrace in the code block by setting `RUST_BACKTRACE=1` in your
environment. E.g. `RUST_BACKTRACE=1 cargo build`.
-->
<details><summary><strong>Backtrace</strong></summary>
<p>

```
thread 'rustc' panicked at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/compiler/rustc_errors/src/lib.rs:1645:9:
Box<dyn Any>
stack backtrace:
   0:     0x7f7803f3c19c - std::backtrace_rs::backtrace::libunwind::trace::haf256adafafbe58d
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x7f7803f3c19c - std::backtrace_rs::backtrace::trace_unsynchronized::hfe1951132ff691c0
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x7f7803f3c19c - std::sys_common::backtrace::_print_fmt::h9a0fe52434930c36
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x7f7803f3c19c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h24adccbf3e1ada83
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/sys_common/backtrace.rs:44:22
   4:     0x7f7803fa289c - core::fmt::rt::Argument::fmt::h18ba555e398addfe
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/core/src/fmt/rt.rs:138:9
   5:     0x7f7803fa289c - core::fmt::write::hbf3ee2d80be74759
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/core/src/fmt/mod.rs:1094:21
   6:     0x7f7803f2e72e - std::io::Write::write_fmt::h095206be3c257ac2
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/io/mod.rs:1714:15
   7:     0x7f7803f3bf85 - std::sys_common::backtrace::_print::hfdc8ddb5f3ddee36
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/sys_common/backtrace.rs:47:5
   8:     0x7f7803f3bf85 - std::sys_common::backtrace::print::h316f264b298c7a30
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/sys_common/backtrace.rs:34:9
   9:     0x7f7803f3f0da - std::panicking::panic_hook_with_disk_dump::{{closure}}::h136bbb963feeea4a
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/panicking.rs:278:22
  10:     0x7f7803f3ec46 - std::panicking::panic_hook_with_disk_dump::hd91a018a982a84f7
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/panicking.rs:305:9
  11:     0x7f7802d0d799 - rustc_driver_impl[2c2bb0da6cfb3826]::install_ice_hook::{closure#0}
  12:     0x7f7803f3f980 - <alloc::boxed::Box<F,A> as core::ops::function::Fn<Args>>::call::hd99b1fe24ccd07b5
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/alloc/src/boxed.rs:2021:9
  13:     0x7f7803f3f980 - std::panicking::rust_panic_with_hook::h1593161995c9c003
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/panicking.rs:733:13
  14:     0x7f78032971f1 - std[b7e5a9c79ea9fbed]::panicking::begin_panic::<rustc_errors[4345527545ed2164]::ExplicitBug>::{closure#0}
  15:     0x7f7803296096 - std[b7e5a9c79ea9fbed]::sys_common::backtrace::__rust_end_short_backtrace::<std[b7e5a9c79ea9fbed]::panicking::begin_panic<rustc_errors[4345527545ed2164]::ExplicitBug>::{closure#0}, !>
  16:     0x7f780322afc6 - std[b7e5a9c79ea9fbed]::panicking::begin_panic::<rustc_errors[4345527545ed2164]::ExplicitBug>
  17:     0x7f780320b344 - <rustc_errors[4345527545ed2164]::HandlerInner>::bug::<alloc[b2a4fa84f81d55c8]::string::String>
  18:     0x7f780320b1d6 - <rustc_errors[4345527545ed2164]::Handler>::bug::<alloc[b2a4fa84f81d55c8]::string::String>
  19:     0x7f780328b20c - rustc_middle[32728bc096d751c7]::util::bug::opt_span_bug_fmt::<rustc_span[9abd768efbccd329]::span_encoding::Span>::{closure#0}
  20:     0x7f780328a00a - rustc_middle[32728bc096d751c7]::ty::context::tls::with_opt::<rustc_middle[32728bc096d751c7]::util::bug::opt_span_bug_fmt<rustc_span[9abd768efbccd329]::span_encoding::Span>::{closure#0}, !>::{closure#0}
  21:     0x7f7803289fd8 - rustc_middle[32728bc096d751c7]::ty::context::tls::with_context_opt::<rustc_middle[32728bc096d751c7]::ty::context::tls::with_opt<rustc_middle[32728bc096d751c7]::util::bug::opt_span_bug_fmt<rustc_span[9abd768efbccd329]::span_encoding::Span>::{closure#0}, !>::{closure#0}, !>
  22:     0x7f780165a400 - rustc_middle[32728bc096d751c7]::util::bug::bug_fmt
  23:     0x7f7801e8d860 - <rustc_borrowck[36619a147d0794a8]::type_check::constraint_conversion::ConstraintConversion>::convert_all
  24:     0x7f7801e8e1c6 - <rustc_borrowck[36619a147d0794a8]::type_check::TypeChecker>::push_region_constraints
  25:     0x7f7801a5ab80 - <rustc_borrowck[36619a147d0794a8]::type_check::TypeChecker>::ascribe_user_type_skip_wf
  26:     0x7f7801a53c90 - rustc_borrowck[36619a147d0794a8]::type_check::type_check
  27:     0x7f7801a4a34f - rustc_borrowck[36619a147d0794a8]::nll::compute_regions
  28:     0x7f7801a24ed5 - rustc_borrowck[36619a147d0794a8]::do_mir_borrowck
  29:     0x7f7801a231e5 - rustc_borrowck[36619a147d0794a8]::mir_borrowck
  30:     0x7f7800d3976e - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>
  31:     0x7f7800d3973e - <rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_span[9abd768efbccd329]::def_id::LocalDefId)>>::call_once
  32:     0x7f7800e80c3d - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::VecCache<rustc_span[9abd768efbccd329]::def_id::LocalDefId, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  33:     0x7f780269094f - rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
  34:     0x7f7801e9cb9f - <rustc_borrowck[36619a147d0794a8]::type_check::TypeChecker>::prove_closure_bounds
  35:     0x7f7801e6b39d - <rustc_borrowck[36619a147d0794a8]::type_check::TypeChecker>::typeck_mir
  36:     0x7f7801a52a01 - rustc_borrowck[36619a147d0794a8]::type_check::type_check
  37:     0x7f7801a4a34f - rustc_borrowck[36619a147d0794a8]::nll::compute_regions
  38:     0x7f7801a24ed5 - rustc_borrowck[36619a147d0794a8]::do_mir_borrowck
  39:     0x7f7801a231e5 - rustc_borrowck[36619a147d0794a8]::mir_borrowck
  40:     0x7f7800d3976e - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>
  41:     0x7f7800d3973e - <rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_span[9abd768efbccd329]::def_id::LocalDefId)>>::call_once
  42:     0x7f7800e80c3d - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::VecCache<rustc_span[9abd768efbccd329]::def_id::LocalDefId, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  43:     0x7f780269094f - rustc_query_impl[c311561e1f786c65]::query_impl::mir_borrowck::get_query_non_incr::__rust_end_short_backtrace
  44:     0x7f780137344e - rustc_mir_transform[2797bcf166cffb39]::mir_drops_elaborated_and_const_checked
  45:     0x7f7800d42a3e - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::mir_drops_elaborated_and_const_checked::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>
  46:     0x7f7800d42a0e - <rustc_query_impl[c311561e1f786c65]::query_impl::mir_drops_elaborated_and_const_checked::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_span[9abd768efbccd329]::def_id::LocalDefId)>>::call_once
  47:     0x7f7800e80c3d - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::VecCache<rustc_span[9abd768efbccd329]::def_id::LocalDefId, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  48:     0x7f780268752f - rustc_query_impl[c311561e1f786c65]::query_impl::mir_drops_elaborated_and_const_checked::get_query_non_incr::__rust_end_short_backtrace
  49:     0x7f7801bf4b63 - rustc_mir_transform[2797bcf166cffb39]::mir_for_ctfe
  50:     0x7f7800dabe4e - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::mir_for_ctfe::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>
  51:     0x7f7800dabe21 - <rustc_query_impl[c311561e1f786c65]::query_impl::mir_for_ctfe::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_span[9abd768efbccd329]::def_id::DefId)>>::call_once
  52:     0x7f7800d38145 - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::DefaultCache<rustc_span[9abd768efbccd329]::def_id::DefId, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  53:     0x7f7802687680 - rustc_query_impl[c311561e1f786c65]::query_impl::mir_for_ctfe::get_query_non_incr::__rust_end_short_backtrace
  54:     0x7f7801279c60 - <rustc_const_eval[9e95d3ca9e32f8a5]::interpret::eval_context::InterpCx<rustc_const_eval[9e95d3ca9e32f8a5]::const_eval::machine::CompileTimeInterpreter>>::load_mir
  55:     0x7f780117f901 - rustc_const_eval[9e95d3ca9e32f8a5]::const_eval::eval_queries::eval_to_allocation_raw_provider
  56:     0x7f7800d8a427 - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 16usize]>>
  57:     0x7f7801c219f7 - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::DefaultCache<rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  58:     0x7f7801c21679 - rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_allocation_raw::get_query_non_incr::__rust_end_short_backtrace
  59:     0x7f780117fe67 - rustc_const_eval[9e95d3ca9e32f8a5]::const_eval::eval_queries::eval_to_allocation_raw_provider
  60:     0x7f7800d8a427 - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_allocation_raw::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 16usize]>>
  61:     0x7f7801c219f7 - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::DefaultCache<rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  62:     0x7f7801c21679 - rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_allocation_raw::get_query_non_incr::__rust_end_short_backtrace
  63:     0x7f780246503a - <rustc_const_eval[9e95d3ca9e32f8a5]::provide::{closure#0} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>)>>::call_once
  64:     0x7f780217a459 - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_valtree::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 24usize]>>
  65:     0x7f780217a408 - <rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_valtree::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>)>>::call_once
  66:     0x7f78022fa0c8 - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::DefaultCache<rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 24usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  67:     0x7f78022f9d5e - rustc_query_impl[c311561e1f786c65]::query_impl::eval_to_valtree::get_query_non_incr::__rust_end_short_backtrace
  68:     0x7f78012eab4c - rustc_middle[32728bc096d751c7]::query::plumbing::query_get_at::<rustc_query_system[1215f1fc07ea991a]::query::caches::DefaultCache<rustc_middle[32728bc096d751c7]::ty::ParamEnvAnd<rustc_middle[32728bc096d751c7]::mir::interpret::GlobalId>, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 24usize]>>>
  69:     0x7f78012ea8eb - <rustc_middle[32728bc096d751c7]::ty::context::TyCtxt>::const_eval_global_id_for_typeck
  70:     0x7f78012ea3c4 - <rustc_middle[32728bc096d751c7]::ty::context::TyCtxt>::const_eval_resolve_for_typeck
  71:     0x7f780145ca87 - <rustc_middle[32728bc096d751c7]::ty::consts::Const>::eval
  72:     0x7f78013e71ff - <rustc_hir_typeck[a1aa762a977bcd63]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  73:     0x7f7801425d54 - <rustc_hir_typeck[a1aa762a977bcd63]::fn_ctxt::FnCtxt>::check_block_with_expected
  74:     0x7f78013e3433 - <rustc_hir_typeck[a1aa762a977bcd63]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args
  75:     0x7f780156899b - <rustc_hir_typeck[a1aa762a977bcd63]::fn_ctxt::FnCtxt>::check_return_expr
  76:     0x7f780156704c - rustc_hir_typeck[a1aa762a977bcd63]::check::check_fn
  77:     0x7f7801555b55 - rustc_hir_typeck[a1aa762a977bcd63]::typeck
  78:     0x7f7800d0b08c - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::typeck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>
  79:     0x7f7800d0b06e - <rustc_query_impl[c311561e1f786c65]::query_impl::typeck::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, rustc_span[9abd768efbccd329]::def_id::LocalDefId)>>::call_once
  80:     0x7f7800e80c3d - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::VecCache<rustc_span[9abd768efbccd329]::def_id::LocalDefId, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  81:     0x7f78026906cf - rustc_query_impl[c311561e1f786c65]::query_impl::typeck::get_query_non_incr::__rust_end_short_backtrace
  82:     0x7f780219be22 - rustc_data_structures[e5ddef51fa833839]::sync::par_for_each_in::<&[rustc_span[9abd768efbccd329]::def_id::LocalDefId], <rustc_middle[32728bc096d751c7]::hir::map::Map>::par_body_owners<rustc_hir_analysis[505bd014cb0020b8]::check_crate::{closure#7}>::{closure#0}>
  83:     0x7f780219ae2a - rustc_hir_analysis[505bd014cb0020b8]::check_crate
  84:     0x7f78021955ca - rustc_interface[b10bd5cb1b971de9]::passes::analysis
  85:     0x7f78021ffd8a - rustc_query_impl[c311561e1f786c65]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[c311561e1f786c65]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 1usize]>>
  86:     0x7f78021ffd79 - <rustc_query_impl[c311561e1f786c65]::query_impl::analysis::dynamic_query::{closure#2} as core[8e2a8f22f3da315]::ops::function::FnOnce<(rustc_middle[32728bc096d751c7]::ty::context::TyCtxt, ())>>::call_once
  87:     0x7f780239b368 - rustc_query_system[1215f1fc07ea991a]::query::plumbing::try_execute_query::<rustc_query_impl[c311561e1f786c65]::DynamicConfig<rustc_query_system[1215f1fc07ea991a]::query::caches::SingleCache<rustc_middle[32728bc096d751c7]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[c311561e1f786c65]::plumbing::QueryCtxt, false>
  88:     0x7f780239b0f7 - rustc_query_impl[c311561e1f786c65]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace
  89:     0x7f7801f798c5 - <rustc_middle[32728bc096d751c7]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[2c2bb0da6cfb3826]::run_compiler::{closure#1}::{closure#2}::{closure#4}, core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>>
  90:     0x7f7801f78ef2 - <rustc_interface[b10bd5cb1b971de9]::interface::Compiler>::enter::<rustc_driver_impl[2c2bb0da6cfb3826]::run_compiler::{closure#1}::{closure#2}, core[8e2a8f22f3da315]::result::Result<core[8e2a8f22f3da315]::option::Option<rustc_interface[b10bd5cb1b971de9]::queries::Linker>, rustc_span[9abd768efbccd329]::ErrorGuaranteed>>
  91:     0x7f7801f71fc8 - std[b7e5a9c79ea9fbed]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[b10bd5cb1b971de9]::util::run_in_thread_pool_with_globals<rustc_interface[b10bd5cb1b971de9]::interface::run_compiler<core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>, rustc_driver_impl[2c2bb0da6cfb3826]::run_compiler::{closure#1}>::{closure#0}, core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>>
  92:     0x7f7801f7174e - <<std[b7e5a9c79ea9fbed]::thread::Builder>::spawn_unchecked_<rustc_interface[b10bd5cb1b971de9]::util::run_in_thread_pool_with_globals<rustc_interface[b10bd5cb1b971de9]::interface::run_compiler<core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>, rustc_driver_impl[2c2bb0da6cfb3826]::run_compiler::{closure#1}>::{closure#0}, core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[8e2a8f22f3da315]::result::Result<(), rustc_span[9abd768efbccd329]::ErrorGuaranteed>>::{closure#1} as core[8e2a8f22f3da315]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  93:     0x7f7803f4a125 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::hb18992c0c074fb04
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/alloc/src/boxed.rs:2007:9
  94:     0x7f7803f4a125 - <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once::ha4998f5f9033c44e
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/alloc/src/boxed.rs:2007:9
  95:     0x7f7803f4a125 - std::sys::unix::thread::Thread::new::thread_start::h3370231efa79af31
                               at /rustc/474709a9a2a74a8bcf0055fadb335d0ca0d2d939/library/std/src/sys/unix/thread.rs:108:17
  96:     0x7f77ff4c76db - start_thread
  97:     0x7f77ff1f061f - clone
  98:                0x0 - <unknown>


rustc version: 1.73.0-nightly (474709a9a 2023-08-03)
platform: x86_64-unknown-linux-gnu

query stack during panic:
#0 [mir_borrowck] borrow-checking `bug::{constant#0}::{closure#0}`
#1 [mir_borrowck] borrow-checking `bug::{constant#0}`
#2 [mir_drops_elaborated_and_const_checked] elaborating drops for `bug::{constant#0}`
#3 [mir_for_ctfe] caching mir of `bug::{constant#0}` for CTFE
#4 [eval_to_allocation_raw] const-evaluating + checking `bug::{constant#0}`
#5 [eval_to_allocation_raw] const-evaluating + checking `bug::{constant#0}`
#6 [eval_to_valtree] evaluating type-level constant
#7 [typeck] type-checking `bug`
#8 [analysis] running analysis passes on this crate
end of query stack

```

</p>
</details>

--- a/tests/ui/borrowck/non-ADT-struct-pattern-box-pattern-ice-121463.rs
+++ b/tests/ui/borrowck/non-ADT-struct-pattern-box-pattern-ice-121463.rs
@@ -1,0 +1,12 @@
+// issue rust-lang/rust#121463
+// ICE non-ADT in struct pattern
+#![feature(box_patterns)]
+
+fn main() {
+    let mut a = E::StructVar { boxed: Box::new(5_i32) };
+    //~^ ERROR failed to resolve: use of undeclared type `E`
+    match a {
+        E::StructVar { box boxed } => { }
+        //~^ ERROR failed to resolve: use of undeclared type `E`
+    }
+}

--- a/tests/ui/borrowck/non-ADT-struct-pattern-box-pattern-ice-121463.stderr
+++ b/tests/ui/borrowck/non-ADT-struct-pattern-box-pattern-ice-121463.stderr
@@ -1,0 +1,21 @@
+error[E0433]: failed to resolve: use of undeclared type `E`
+  --> $DIR/non-ADT-struct-pattern-box-pattern-ice-121463.rs:6:17
+   |
+LL |     let mut a = E::StructVar { boxed: Box::new(5_i32) };
+   |                 ^
+   |                 |
+   |                 use of undeclared type `E`
+   |                 help: a trait with a similar name exists: `Eq`
+
+error[E0433]: failed to resolve: use of undeclared type `E`
+  --> $DIR/non-ADT-struct-pattern-box-pattern-ice-121463.rs:9:9
+   |
+LL |         E::StructVar { box boxed } => { }
+   |         ^
+   |         |
+   |         use of undeclared type `E`
+   |         help: a trait with a similar name exists: `Eq`
+
+error: aborting due to 2 previous errors
+
+For more information about this error, try `rustc --explain E0433`.

--- a/tests/ui/const-generics/generic_const_exprs/cannot-convert-refree-ice-114463.rs
+++ b/tests/ui/const-generics/generic_const_exprs/cannot-convert-refree-ice-114463.rs
@@ -1,0 +1,10 @@
+// issue: rust-lang/rust#114463
+// ICE cannot convert `ReFree ..` to a region vid
+#![feature(generic_const_exprs)]
+//~^ WARN the feature `generic_const_exprs` is incomplete and may not be safe to use and/or cause compiler crashes
+fn bug<'a>() {
+    [(); (|_: &'a u8| (), 0).1];
+    //~^ ERROR cannot capture late-bound lifetime in constant
+}
+
+pub fn main() {}

--- a/tests/ui/const-generics/generic_const_exprs/cannot-convert-refree-ice-114463.stderr
+++ b/tests/ui/const-generics/generic_const_exprs/cannot-convert-refree-ice-114463.stderr
@@ -1,0 +1,19 @@
+warning: the feature `generic_const_exprs` is incomplete and may not be safe to use and/or cause compiler crashes
+  --> $DIR/cannot-convert-refree-ice-114463.rs:3:12
+   |
+LL | #![feature(generic_const_exprs)]
+   |            ^^^^^^^^^^^^^^^^^^^
+   |
+   = note: see issue #76560 <https://github.com/rust-lang/rust/issues/76560> for more information
+   = note: `#[warn(incomplete_features)]` on by default
+
+error: cannot capture late-bound lifetime in constant
+  --> $DIR/cannot-convert-refree-ice-114463.rs:6:16
+   |
+LL | fn bug<'a>() {
+   |        -- lifetime defined here
+LL |     [(); (|_: &'a u8| (), 0).1];
+   |                ^^
+
+error: aborting due to 1 previous error; 1 warning emitted
+

--- a/tests/ui/consts/const_refs_to_static-ice-121413.rs
+++ b/tests/ui/consts/const_refs_to_static-ice-121413.rs
@@ -1,0 +1,16 @@
+// ICE: ImmTy { imm: Scalar(alloc1), ty: *const dyn Sync } input to a fat-to-thin cast (*const dyn Sync -> *const usize
+// or with -Zextra-const-ub-checks: expected wide pointer extra data (e.g. slice length or trait object vtable)
+// issue: rust-lang/rust#121413
+//@ compile-flags: -Zextra-const-ub-checks
+// ignore-tidy-linelength
+#![feature(const_refs_to_static)]
+const REF_INTERIOR_MUT: &usize = {
+    static FOO: Sync = AtomicUsize::new(0);
+    //~^ ERROR failed to resolve: use of undeclared type `AtomicUsize`
+    //~| WARN trait objects without an explicit `dyn` are deprecated
+    //~| ERROR the size for values of type `(dyn Sync + 'static)` cannot be known at compilation time
+    //~| ERROR the size for values of type `(dyn Sync + 'static)` cannot be known at compilation time
+    //~| WARN this is accepted in the current edition (Rust 2015) but is a hard error in Rust 2021!
+    unsafe { &*(&FOO as *const _ as *const usize) }
+};
+pub fn main() {}

--- a/tests/ui/consts/const_refs_to_static-ice-121413.stderr
+++ b/tests/ui/consts/const_refs_to_static-ice-121413.stderr
@@ -1,0 +1,46 @@
+error[E0433]: failed to resolve: use of undeclared type `AtomicUsize`
+  --> $DIR/const_refs_to_static-ice-121413.rs:8:24
+   |
+LL |     static FOO: Sync = AtomicUsize::new(0);
+   |                        ^^^^^^^^^^^ use of undeclared type `AtomicUsize`
+   |
+help: consider importing this struct
+   |
+LL + use std::sync::atomic::AtomicUsize;
+   |
+
+warning: trait objects without an explicit `dyn` are deprecated
+  --> $DIR/const_refs_to_static-ice-121413.rs:8:17
+   |
+LL |     static FOO: Sync = AtomicUsize::new(0);
+   |                 ^^^^
+   |
+   = warning: this is accepted in the current edition (Rust 2015) but is a hard error in Rust 2021!
+   = note: for more information, see <https://doc.rust-lang.org/nightly/edition-guide/rust-2021/warnings-promoted-to-error.html>
+   = note: `#[warn(bare_trait_objects)]` on by default
+help: if this is an object-safe trait, use `dyn`
+   |
+LL |     static FOO: dyn Sync = AtomicUsize::new(0);
+   |                 +++
+
+error[E0277]: the size for values of type `(dyn Sync + 'static)` cannot be known at compilation time
+  --> $DIR/const_refs_to_static-ice-121413.rs:8:17
+   |
+LL |     static FOO: Sync = AtomicUsize::new(0);
+   |                 ^^^^ doesn't have a size known at compile-time
+   |
+   = help: the trait `Sized` is not implemented for `(dyn Sync + 'static)`
+
+error[E0277]: the size for values of type `(dyn Sync + 'static)` cannot be known at compilation time
+  --> $DIR/const_refs_to_static-ice-121413.rs:8:24
+   |
+LL |     static FOO: Sync = AtomicUsize::new(0);
+   |                        ^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
+   |
+   = help: the trait `Sized` is not implemented for `(dyn Sync + 'static)`
+   = note: constant expressions must have a statically known size
+
+error: aborting due to 3 previous errors; 1 warning emitted
+
+Some errors have detailed explanations: E0277, E0433.
+For more information about an error, try `rustc --explain E0277`.

